### ISSUE TITLE
Deprecate the paging model

### ISSUE THREAD

**Issue by azaroth42** [IIIF/api#1343]
In the Presentation API v2.1, there are two resources that allow paging: AnnotationLists and Collections.  In v3.0, AnnotationLists are pushed to the equivalent construction in the Web Annotation Model, an Annotation Collection Page that follows exactly the same model (Activity Streams 2.0 Ordered Paged Collection). This pattern was not available in the Open Annotation model, and its inclusion in the W3C Technical Recommendation drew from the use in IIIF amongst other sources. As such, we no longer need to specify the paging model for that construction separately.

Collections already allow, in the same way as Ranges, a recursive inclusion to arbitrary depth.  The semantics of a Collection are simply:
> Collections allow easy advertising and browsing of the manifests in a hierarchical structure, potentially with its own descriptive information.

Which allows already for paging as a mechanism for browsing the set of manifests -- by including page-like collections within a master collection, with an appropriate label. Collections and Manifests in 2.1 (and later) may also be interleaved, unlike with the strict paging model where the page is a transmission artifact rather than a persistent entity.

Given that the use cases that necessitated the introduction of the paging section appear to be met, and the Collection Pages are inconsistent with the ActivityStreams pages (in that they are Collections, not CollectionPages), I propose to deprecate the paging model in Presentation API 3.0

**Comment by jpstroop** [comment]
My first reaction is that this overloads collections in a negative way--they'd be both for sematic (e.g. thematic) grouping as well as arbitrary chunking (paging), and it eliminates the ability of the client to choose the page size.

**Comment by azaroth42** [comment]
That's exactly the situation we have now with Collections! 

Collections are used as both the actual "collection" and the page, per http://iiif.io/api/presentation/2.1/#paging.  And we don't allow the client to choose the page size.

**Comment by tomcrane** [comment]
Can we point to two implementations of collection paging in the wild?

**Comment by tomcrane** [comment]
AV Community call - no immediate blockers to deprecation, but need to seek further input on iiif-discuss

**Comment by workergnome** [comment]
I'm a +1 for depreciating.

**Comment by ahankinson** [comment]
Referring, in part, to Tom's e-mail on IIIF-Discuss:

Our 'collections' are created dynamically by performing a Solr query and re-formatting the results. In some cases, we have a very large number of references:
https://iiif.bodleian.ox.ac.uk/iiif/collection/All

In some cases, we have a very small number:
https://iiif.bodleian.ox.ac.uk/iiif/collection/Merton

And, in some cases, we have a number somewhere in-between:
https://iiif.bodleian.ox.ac.uk/iiif/collection/Polonsky

So I'm not sure what 'human-sized' chunks we would break our IIIF Collections into, since the materials in ours (and I suppose, many others') collections can be quite extensive. If someone wants to find all manifests for our Western Manuscripts, what would be the appropriate human-sized chunk we should publish it as?

From what I remember from Toronto, ActivityStreams were seen as a solution for discovering newly-published resources, rather than a "queryable" result suitable for all use cases covered by Collections.

I'm intrigued by Rob's initial comment, but would probably have to see a mockup for it to completely make sense. (Sorry, my W3Cese is a bit rusty. 😄 )

**Comment by azaroth42** [comment]
You could break them into arbitrary slices of 50 (or whatever page size you want to provide). The end result is the same -- there's a collection of collections, where the children are the "pages", we just don't need the paging mechanisms that ActivityStreams provides.

The only situation I can think of where paging is useful is when there are so many pages that they're not realistically enumerable in a single response. E.g. if you have a single collection of 10 million objects, and you want only 10 items per page, you need to include a million URLs as the `items` of the top collection.  I consider this as a pathological edge case.

150k entries, at 50 entries per page, with 60 characters per page URL is only 180k characters. There are many manifests much longer than that!

**Comment by tomcrane** [comment]
What needs, and whose needs, are we serving by publishing Collections? Right now, sometimes those chunks of JSON drive user interface, sometimes they are consumed by scripts.

* I want to browse your digitised collections in a nice UI
* I want to deliver manifest-level search results to a user interface 
* I want to collect a set of objects associated with an exhibition, or an article, or my bookmarks
* I want to point my harvesting code at some endpoint and discover all your IIIF resources

Publishing collections like https://wellcomelibrary.org/service/collections/ or https://iiif.bodleian.ox.ac.uk/iiif/collection/All is often used for the last of these, and gives very mixed results for the first. Sometimes small nodes like https://wellcomelibrary.org/service/collections/topics/Cats/, sometimes the semi-pathological edge cases with tens of thousands of manifests. Right now, the same top level endpoint is used by people who want to crawl your stuff, and by people who open and explore The Bodleian or Wellcome in _FromThePage_, looking for something to transcribe.

There are no semantics available to a crawler to know why a collection hierarchy is split up the way it is. So whether it follows paging or traverses a collection tree to get everything doesn't make a difference. Which leaves Collections for Presentation use cases. These could still involve very large sets; sometimes that is appropriate in a UI (e.g., Zegami), sometimes not (a selection for rendering as a highlight reel on the homepage). If, say, you wanted your Polonsky collection to be user-friendly when opened in Mirador's initial selection UI mode (assuming support for navigating nested collections as per various plugins), you'd break it into child collections with friendly labels, which would give a better user experience than arbitrary paging (which would not be under the client's control anyway):

`"label": "Douce 1 - Douce V 245 (123 items)"`

...and even if you weren't in a position to split unevenly based on some criteria, the UI would be no worse than if the client had nothing but paging to fall back on to generate a human-sized UI.

**Comment by azaroth42** [comment]
And (probably this is obvious, but to be clear) the last bullet is what the Discovery work aims to solve with ActivityStreams, rather than with Collections, for exactly that reason.

And to clarify the pathological case ... the reason it works with paging is because the pages are an open-ended forward-linked list.  The collection references the first page, the first page references the second page, which references the next page and so on.  There's no requirement for the system to know the number of pages, nor all of their URIs at a single point in time.  This would be important for predictive systems that make an educated guess at the number of results (like Google) rather than deterministic systems that do the calculation for real. I don't think there are any real world use cases that fall under the pathological case in the IIIF universe, or I wouldn't be proposing removing paging :)

**Comment by tomcrane** [comment]
Yeah, I should have made that clearer; _right now_ we use Collections for all these use cases; going forward we are left with Presentation use cases. For humans.

**Comment by ahankinson** [comment]
I guess I'm primarily asking for clarification of the issue, and what the alternative might actually look like. Right now, it's a bit too abstract for me to fully understand the implications of deprecating it when the current Collections functionality has proven useful. (Nevermind the fact that we didn't get around to implementing paged collections anyway! 😬 )

So could someone mock out some JSON that would illustrate how the proposed alternatives would look? Right now it's all a bit abstract.

**Comment by azaroth42** [comment]
There are currently two possible patterns:

Enumerated List:
```
{
  "@id": "top-collection",
  "@type": "sc:Collection",
  "members": [
     {"@id": "sub-collection1", "@type": "sc:Collection"}, 
     {"@id": "sub-collection2", @type": "sc:Collection"},
     ...
  ]
}
```

Paged (forward linked list):
```
{
  "@id": "top-collection",
  "@type": "sc:Collection",
  "first":  {"@id": "sub-collection1", "@type": "sc:Collection"}
}
```
And then that collection has a `next` property that references `sub-collection2`, and so on.

This proposal would remove the second possibility and, therefore, require the first.

**Comment by ahankinson** [comment]
So does this mean:

```
{
  "@id": "sub-collection1",
  "@type": "sc:Collection",
  "members": [
    {"@id": "manifest1", "@type": "sc:Manifest"},
    {"@id": "manifest2", "@type": "sc:Manifest"},
    ... x50
  ]
}
```

If so:

1. Can `@type=sc:Manifest` and `sc:Collection` be mixed in the `members` array? (I presume so)
2. If 1, does that put extra burden on consumer applications, and leave the possibility of undefined behaviours open? What is expected behaviour if I encounter a members list of mixed collections and manifests?
3. If 1, What's the 'human' limit for where to break top-level collections into sub-collections? 10? 50? 100? If I have a collection of just 4 manifests, I wouldn't be expected to publish a top-level collection with a sub-collection, would I?
4. Would I ever need to have more than two-level `sc:Collection` layers? Top level, with second level enumerated in the top level, broken into as many as needs be? Or would we limit the number of `members` to be also 'human', meaning, say, 100 top level collections, each with 100 sub-level collections, which then list 100 manifests? (In the case of the "All" collection, this would only cover 1m manifests, and many orgs have more than this, so we could imagine a third level.)

I'm not trying to be difficult, honest! It seems that the 'paging' model ("You're on page N of M, with L total results") is quite well understood. Simplifying this by making the pages into sub-collections seems like we gain spec simplicity at the expense of introducing a new paging pattern, with associated human costs of teaching and communicating this new pattern.

**Comment by azaroth42** [comment]
1. Yes.
2. The expectation, given it's for navigation, is that the client presents the members to the user to view ... probably by allowing expansion of the collections and selection of the manifests.
3. Entirely up to the publisher! If you get bored scrolling scrolling scrolling ... probably it's too big. 
4. Also entirely up to the publisher :)

**Comment by azaroth42** [comment]
Eds agree that cookbook is the right place for collection paging examples.

**Comment by zimeon** [comment]
Closed by #1406

**Comment by mixterj** [comment]
+1 for deprecating the paging model
